packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,385 @@
+      const { membershipTestResults } =
+        testResultCount5000Rate01 as TestResultType;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bits.bitmap || ''),
+        bits.padding || 0,
+        hashCount || 0
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const backendMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
+        const mightContain = bloomFilter.mightContain(documentPrefix + i);
+        expect(mightContain).to.equal(backendMembershipResult);
+      }
+    });
+    it('mightContain result for 5000 documenta with 0.0001 false positive rate', () => {
+      const { bits, hashCount } = testDataCount5000Rate0001 as TestDataType;
+      const { membershipTestResults } =
+        testResultCount5000Rate0001 as TestResultType;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bits.bitmap || ''),
+        bits.padding || 0,
+        hashCount || 0
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const backendMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
+        const mightContain = bloomFilter.mightContain(documentPrefix + i);
+        expect(mightContain).to.equal(backendMembershipResult);
+      }
+    });
+    it('mightContain result for 50000 documents with 1 false positive rate', () => {
+      const { bits, hashCount } = testDataCount50000Rate1 as TestDataType;
+      const { membershipTestResults } =
+        testResultCount50000Rate1 as TestResultType;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bits.bitmap || ''),
+        bits.padding || 0,
+        hashCount || 0
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const backendMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
+        const mightContain = bloomFilter.mightContain(documentPrefix + i);
+        expect(mightContain).to.equal(backendMembershipResult);
+      }
+    });
+    it('mightContain result for 50000 documents with 0.01 false positive rate', () => {
+      const { bits, hashCount } = testDataCount50000Rate01 as TestDataType;
+      const { membershipTestResults } =
+        testResultCount50000Rate01 as TestResultType;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bits.bitmap || ''),
+        bits.padding || 0,
+        hashCount || 0
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const backendMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
+        const mightContain = bloomFilter.mightContain(documentPrefix + i);
+        expect(mightContain).to.equal(backendMembershipResult);
+      }
+      //Extend default timeout(2000)
+    }).timeout(10_000);
+
+    it('mightContain result for 50000 documents with 0.0001 false positive rate', () => {
```

Comment:
files are too many and the hash calculation is time consuming. This is failing in github check even with extended time out. Should I extend it more?